src/delaunay_core/dcel.rs

```diff
@@ -133,6 +133,15 @@ impl<V, DE, UE, F> Dcel<V, DE, UE, F> {
         self.faces.truncate(1); // Keep outer face
     }
 
+    pub fn get_vertex(&self, index: usize) -> Option<VertexHandle<V, DE, UE, F>> {
+        match self.vertices.get(index) {
```

Comment:
Would you mind to...

 1. Change the type from `usize` to `FixedVertexHandle`? I think that's more consistent and avoids passing in the wrong index type.
 2. Write this without `vertices.get(...)` - we don't really require accessing any memory yet, we just need to validate the index. I believe `(handle.index() < self.vertices.len()).then(|| self.vertex(handle))` should work and be idiomatic.